src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -566,6 +570,27 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
 
     StencilConstructor.startBody();
 
+    // Add static asserts to check halos against extents
+    StencilConstructor.addComment("Check if extents do not exceed the halos");
+    auto exts = (*stencilInstantiation->getStencils()[stencilIdx]).computeEnclosingAccessExtents();
```

Comment:
I would not use auto here, but the real type, for readability. Specially since the name indicates this could be of class `Extents`

Reply:
fixed with https://github.com/MeteoSwiss-APN/dawn/pull/96/commits/6e09d5205ce94885f57ba4c9620488826e682232